### ISSUE TITLE
Incorrect Refactoring: Self referential comprehension

### ISSUE THREAD

**Issue by Wbec** [sourcery-ai/sourcery#67]
### Issue description or question
Sourcery suggests refactoring:
```
def test():
    foo = []
    for i in range(10):
        foo.append(len(foo))
```
into
```    
def test():
    foo = [len(foo) for _ in range(10)]
```
which raises a value error when run, since foo is not defined inside the comprehension.

Note that this has been simplified, the original code snipit having the issue was:
```
unq_names = {}
for i, original in enumerate(items):
    unq_names[original] = unq_names.get(original, []) + [i]
```
which builds a dictionary of the positions of duplicate names in a single pass over the input.

### Sourcery Version
0.7.5

### Code editor or IDE name and version

Pycharm Professional v2020.2

### OS name and version

Ubuntu 20.04 LTS

**Comment by Hellebore** [comment]
Thanks for the report! Looks like a definite bug - will investigate.

**Comment by Hellebore** [comment]
Resolved with release of 0.7.9
